File: app/src/main/java/com/example/lockdowndpc/ui/BlockedBrowserActivity.java
package com.example.lockdowndpc.ui;

import android.app.Activity;
import android.graphics.Color;
import android.os.Bundle;
import android.view.Gravity;
import android.view.ViewGroup;
import android.widget.LinearLayout;
import android.widget.TextView;

public final class BlockedBrowserActivity extends Activity {
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        getWindow().setStatusBarColor(Color.WHITE);

        LinearLayout root = new LinearLayout(this);
        root.setOrientation(LinearLayout.VERTICAL);
        root.setGravity(Gravity.CENTER);
        root.setPadding(dp(32), dp(32), dp(32), dp(32));
        root.setLayoutDirection(LinearLayout.LAYOUT_DIRECTION_RTL);

        TextView icon = new TextView(this);
        icon.setText("🛡️");
        icon.setTextSize(56);
        icon.setGravity(Gravity.CENTER);
        root.addView(icon, matchWrap());

        TextView title = new TextView(this);
        title.setText("הגישה לאינטרנט חסומה");
        title.setTextSize(26);
        title.setTextColor(Color.rgb(25, 30, 38));
        title.setGravity(Gravity.CENTER);
        title.setPadding(0, dp(20), 0, dp(8));
        root.addView(title, matchWrap());

        TextView message = new TextView(this);
        message.setText("לא ניתן לפתוח דפדפנים או קישורי אינטרנט במכשיר הזה.");
        message.setTextSize(17);
        message.setTextColor(Color.rgb(80, 88, 101));
        message.setGravity(Gravity.CENTER);
        root.addView(message, matchWrap());

        setContentView(root);
    }

    private ViewGroup.LayoutParams matchWrap() {
        return new ViewGroup.LayoutParams(ViewGroup.LayoutParams.MATCH_PARENT, ViewGroup.LayoutParams.WRAP_CONTENT);
    }

    private int dp(int value) {
        return Math.round(value * getResources().getDisplayMetrics().density);
    }
}

File: app/src/main/java/com/example/lockdowndpc/ui/BlockedBrowserActivity.kt
package com.example.lockdowndpc.ui

import android.graphics.Color
import android.os.Bundle
import androidx.activity.SystemBarStyle
import androidx.activity.compose.setContent
import androidx.activity.enableEdgeToEdge
import androidx.appcompat.app.AppCompatActivity
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.size
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.rounded.PublicOff
import androidx.compose.material3.Card
import androidx.compose.material3.CardDefaults
import androidx.compose.material3.Icon
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.res.stringResource
import androidx.compose.ui.text.style.TextAlign
import androidx.compose.ui.unit.dp
import com.example.lockdowndpc.R
import com.example.lockdowndpc.ui.theme.LockdownTheme

/**
 * Safe destination for HTTP and HTTPS intents while browser access is blocked.
 *
 * `AppCompatActivity` is the base class so this screen honours the persisted
 * display language on API levels below 33, like the rest of the console.
 */
class BlockedBrowserActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        enableEdgeToEdge(
            statusBarStyle = SystemBarStyle.light(Color.TRANSPARENT, SYSTEM_BAR_DARK_SCRIM),
            navigationBarStyle = SystemBarStyle.light(Color.TRANSPARENT, SYSTEM_BAR_DARK_SCRIM),
        )
        setContent {
            LockdownTheme { BrowserBlockedScreen() }
        }
    }
}

@Composable
private fun BrowserBlockedScreen() {
    LockdownScreen(
        title = stringResource(R.string.browser_blocked_title),
        subtitle = stringResource(R.string.browser_blocked_subtitle),
    ) {
        Card(
            modifier = Modifier.fillMaxWidth(),
            shape = MaterialTheme.shapes.large,
            colors = CardDefaults.cardColors(
                containerColor = MaterialTheme.colorScheme.primaryContainer,
                contentColor = MaterialTheme.colorScheme.onPrimaryContainer,
            ),
        ) {
            Column(
                horizontalAlignment = Alignment.CenterHorizontally,
                modifier = Modifier
                    .fillMaxWidth()
                    .padding(horizontal = 24.dp, vertical = 32.dp),
            ) {
                Icon(
                    imageVector = Icons.Rounded.PublicOff,
                    contentDescription = null,
                    modifier = Modifier.size(48.dp),
                )
                Spacer(Modifier.height(18.dp))
                Text(
                    text = stringResource(R.string.browser_blocked_message),
                    style = MaterialTheme.typography.bodyLarge,
                    textAlign = TextAlign.Center,
                )
            }
        }
    }
}
